### PR TITLE
Docs: Contributing to Weblate documentation reworked

### PR DESCRIPTION
**Pull request by comradekingu**
Don't write documentation because other people don't read it; rather because they _resorted_ to reading it.

Could have a UI section, but that is some parts graphics, and partway code, and I don't know what to tell anyone about doing it because I haven't done any of it in Weblate.

Is there anything on how to join the various teams? Is there one for graphics?

### CODE REVIEW

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:7]: Two sentences doing what the sidebar does.
Then a table of contents doing it yet again.
Then (if there wasn't so much text), headlines that can actually be browsed to that effect. Then "in depth" guide links making it all hollow.

Make sure the first try sticks, and then get fancy.

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:10]: (is out)

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:11]: (is out)

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:3]:
```suggestion
Non-code contributions
```
In the interest of keeping the sidemenu short, and keeping code and code approaches in that section.

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:37]: should be "considerate".

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:30]: Instead of all the "this happened once". Let's not make reading about it a hurdle to overcome for every new contributor.

File: docs/contributing/index.rst
Comment (comradekingu) [inline docs/contributing/index.rst:15]: Are documentation issues tagged anything other than "ux"?

Comment (orangesunny) [reply docs/contributing/index.rst:15]: Yes, there’s a label `documentation`. Thanks for the head‘s up, we forgot to use it before.

There is no need to hard-code assigning link into the docs, but we have a team [`reviewers`](https://github.com/orgs/WeblateOrg/teams/reviewers) you are a member of.

**Comment by orangesunny** [comment]
> Is there anything on how to join the various teams? Is there one for graphics?

There is no graphics team at the moment, but anything that’s going out is open so you can check and comment.

File: docs/contributing/index.rst
Comment (orangesunny) [reply docs/contributing/index.rst:30]: Obstacles during merging docs' PRs are often. Also, many clients asked me for this so we have the rules that can be improved, not removed.

I am sorry, but I cannot let you do the documentation merges. You changed the meaning of the sentences many times. PRs also contained changes that introduced descriptions not according to the reality of the app. Everybody can make mistakes so we have multi-level reviews :)

File: docs/contributing/index.rst
Comment (orangesunny) [reply docs/contributing/index.rst:37]: Thank you for the opinion.  I used `thoughtful` intentionally.

If we look at
https://www.merriam-webster.com/dictionary/thoughtful
I chose it especially because of the part 1b and even more 2b.

File: docs/contributing/index.rst
Comment (nijel) [inline docs/contributing/index.rst:24]: I don't see a reason for including these links - there is nothing to be done there (one is finished, just cannot be released for whatever reasons on their side and one is on hold until we get more starts).

File: docs/contributing/index.rst
Comment (nijel) [inline docs/contributing/index.rst:21]: This overlaps with https://docs.weblate.org/en/latest/contributing/about.html#project-logos

File: docs/contributing/index.rst
Comment (nijel) [reply docs/contributing/index.rst:7]: Reader should be able to follow the documentation from the text. The sidebar is nice, but hardly usable on mobile and having links in the text is way easier to navigate in this case.

Compare:
image image

File: docs/contributing/index.rst
Comment (nijel) [inline docs/contributing/index.rst:28]: We definitely need a documentation how to contribute to the documentation. Several people have asked us for a guidance here and this is the initial version of that. The motivation is to make it both easy to contribute and easy to review and merge. Huge pull requests which take weeks to review are not good for either side - the contributor feels bad because there is need for continuous rewriting and reviewers get lost in what has part has been approved and which not.

File: docs/contributing/index.rst
Comment (nijel) [reply docs/contributing/index.rst:3]: This page was a general contributing landing page. I don't think removing the code part is a good idea.

Also, this would change the label of all references to this section and that would make it harder to understand - "See also: Contributing to Weblate" is clear, while "See also: Non-code contributions" is not.

**Review by nijel — CHANGES_REQUESTED** [review]
Thanks for your contribution, but I don't think this is something we can merge.

**Comment by orangesunny** [comment]
While this isn’t mergeable, I created https://github.com/WeblateOrg/weblate/pull/5697 with some improvements there. Feel free to drop your feedback in the comments.

File: docs/contributing/index.rst
Comment (comradekingu) [reply docs/contributing/index.rst:24]: It is a fantastic example of how to improve Weblate in other venues. Just adding a star is maybe the best example of the simplest thing to do, with somewhat of an impact.

File: docs/contributing/index.rst
Comment (comradekingu) [reply docs/contributing/index.rst:21]: This PR removes layers and layers of overlap. The real question is why pointing to the logos suggests they can be worked on, or details why. Listing that in "about" is usually done for people to use them (for which there are no stated terms). It is sort of a media-pack, only there is no write-up to use. (Though that fell out of fashion with SEO-ing, some basic history would do nicely. (Like linking to the Wikipedia article.) That is a good idea for writing _less_ documentation, in a more honest way.

File: docs/contributing/index.rst
Comment (nijel) [reply docs/contributing/index.rst:24]: Sorry, but I don't think it's reasonable to expect people to star a repo because they read an issue which is linked in the documentation. Starting might be the simplest thing to do, but figuring it out this way is definitely not.

File: docs/contributing/index.rst
Comment (comradekingu) [reply docs/contributing/index.rst:3]: What on earth is a "landing page" for contributions, if not something that wastes attention, short of actually conveying any useful info. It is a suggestion, because it may break assumptions.

There is no different way of contributing translations for Weblate, translating other projects also helps the general esteem of Weblate, but also the shared TM (for what is actually works between Weblate and other projects on Hosted).

As such, a "Contributions" category is a mistake. This all belongs in the general user docs, with Hosted Weblate/Weblate used only as an example. That way other projects can integrate it in their guidelines. Bonus points for placeholders instead of hardcoded Hosted Weblate/Weblate, but alas.

File: docs/contributing/index.rst
Comment (comradekingu) [reply docs/contributing/index.rst:28]: @nijel I will grant you both it needs doing (with some level of urgency below that of fixing the access docs being a mess and appearing first for translation (without being marked up with terminology or contextualised), _and_ that people ask about it.

Instead of something that makes it easy to contribute, you have made something that I am walking away from. The rules in this are moronic and reactionary. I value your time, and I wish you the best of luck.

As of this year I can actually call myself a professional libre software documentation writer. It isn't lucrative, but I mention it so that with _every_ bit of experience and conviction I have, I can say you have the wrong ideas, and end up in downfall of the commons on every level. Fat-Zer has qualities that I don't. I hope you can make him the head of the documentation. I can offer to pay him to do so.

https://github.com/WeblateOrg/weblate/pull/5566#issuecomment-801842143
Here you have an example of the disconnect between being asked how to do something, and the idea that it makes one capable of answering.

File: docs/contributing/index.rst
Comment (orangesunny) [reply docs/contributing/index.rst:3]: @comradekingu

If two people from the core Weblate team are telling you something, please consider it has some reason instead of telling everyone it’s a mistake or "what on the Earth...".

These follow-up PRs are not counters, please don’t take it personally. It’s an evolution. If creating a follow-up PR makes more sense than correcting something hardly mergeable, it’s not a personal attack. It’s not ideal, but it’s the best way, sometimes. The original author can still collaborate and give it their best.

Comment (comradekingu) [reply docs/contributing/index.rst:3]: The distinction of who is on the core team is meaningless to me. I respect the work of people that do good work.
Two people from the core Weblate team is telling me what exactly? Second time you pulled rank.
What is the reason there? I am telling the world at large I don't think changing to "non-code contribution" is a change without its issues. Hence "suggested change".
Also, I can get links to actually useful resources back in to facilitate phone-readers. I OTOH read it on a 3860 px vertical screen, and the sidemenu has enough entries to where it almost scrolls…

Speaking of, your guide is _a lot_ longer, that is _a lot_ worse for mobile-readers. It is worse for blind people, it is worse for people looking to find and use info. There is no overview to be had.

You are hung up in language, rather than arguing the point, which is "short of" part.
There are reasons here. Additionally, you have replicated info found elsewhere, but reworded it. There is nothing personal there, nor am I taking it that way. It is not an evolution. You going your own way is the opposite.
That you work in corrections I make is not an evolution of the origin. It is a flat refusal to entertain a discussion you invited, for what you said was a basis to work on.

Sometimes shutting down a PR is the best option, yes. When you yolo in a page that is your own, it would have been.
Evidently, your version is full of mistakes. And still is.
The original author is you because you decided to be with commit access, not because I respect your work.
It can sit there and be laughable. Nothing better than seeing my own language tarnished, in a further blight to the resource I worked on. What's not to like? "Landing page" in the documentation awful, newest blogpost basic spelling errors.
Why are you in charge of that?

How do you make it past whatever workflow you have with that? Why are you beyond the actual critique you don't even care to offer others?

I can possibly "check and comment", but I didn't even know it existed.

Put, it, on, review, you are the public image of the one libre translation platform we have.
There is a "reviewers" team. Don't know if it is called "documentation", and it would be a welcomed change if either started to be part of what that group actually does.

Your edit also says donations are used to host other projects. That is not what it says on the actual donation page.
Nor is it actually something that belongs in "Contributing to Weblate". The psychology is sub-par with that angle.
Not my call to decide how funds are used, or what to merge, but I used my time to hone the entire string-base of the website.

"While this isn’t mergeable", means _what_ exactly, seeing as you wrote it as you closed this PR?

Comment (nijel) [reply docs/contributing/index.rst:3]: > Also, I can get links to actually useful resources back in to facilitate phone-readers.

All I said is that it makes navigating the documentation for some readers worse. It can be certainly improved (anything can be improved), but it cannot be merged as it is.

This PR exactly shows reasons why we've added a guideline on contributing to the documentation, which you have removed. This PR contains parts which could be more or less merged right now (and @orangesunny extracted those to https://github.com/WeblateOrg/weblate/pull/5697), parts which need additional work and parts which are not acceptable. Doing smaller pull requests makes it easier to review, merge and improve the documentation in smaller steps.

Comment (orangesunny) [reply docs/contributing/index.rst:3]: @comradekingu
I am sorry to see that you switched to a way in which you are picking anything to call it bad, worse, a mistake, using strong language. This is something we can hardly work with. I will be happy if you switch back to the constructive way I know you can navigate pretty well.

Please drop me a message on Wire, or in an e-mail, let’s talk and clear the air so we can understand each other better and continue to cooperate productively.

To summarize this issue from my point of view:
1. I wrote the https://github.com/WeblateOrg/weblate/pull/5658
   - to include what Weblate users and customers were asking for.
   - to prevent the issues we bumped into while working on *Access* article from happening again.
   - merge it together with @nijel. Once it’s released, everybody can read it, respect it, and suggest a change if needed.
2. You read it and suggested a change here (https://github.com/WeblateOrg/weblate/pull/5688), changing the structure, purpose, and deleting the whole new part.
3. As it is a pull "request", we checked it with @nijel and realized it’s not mergeable. So ⬇️ 
4. I tried to use all of your good parts from here in https://github.com/WeblateOrg/weblate/pull/5697 and ask you to comment there so we can make it good together. And you start down-talking it and saying you are not able to work with that.

Again, I am sorry to read you are not happy with it and I am sure we can find a way to cooperate. Let’s talk.

I believe that we will write and link the materials that will help us and newcomers to orient in the docs’ contribution, show how and which syntaxes to use, and make docs writing easier and more effective for everyone.

**Pull request closed without merging**
